### PR TITLE
Some Cleanup in BlobStoreRepository

### PR DESCRIPTION
**Pull request by original-brownbear**
* Extracted from #42833:
  * Dry up index and shard path handling
  * Shorten XContent handling

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java [hunk 1]
```diff
@@ -540,8 +533,19 @@ public MetaData getSnapshotGlobalMetaData(final SnapshotId snapshotId) {
...
+
+    private BlobPath indicesPath() {
+        return basePath().add("indices");
+    }
+
+    private BlobContainer indexContainer(IndexId indexId) {
+        return blobStore().blobContainer(indicesPath().add(indexId.getId()));
+    }
+
+    private BlobContainer shardContainer(IndexId indexId, ShardId snapshotShardId) {
```
Comment (tlrx) [inline server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java]: Nit: snapshotShardId -> shardId

File: server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java [hunk 2]
```diff
@@ -685,19 +688,11 @@ protected void writeIndexGen(final RepositoryData repositoryData, final long rep
...
-                builder.close();
-            }
-            snapshotsBytes = bStream.bytes();
-        }
         // write the index file
         final String indexBlob = INDEX_FILE_PREFIX + Long.toString(newGen);
         logger.debug("Repository [{}] writing new index generational blob [{}]", metadata.name(), indexBlob);
-        writeAtomic(indexBlob, snapshotsBytes, true);
+        writeAtomic(
+            indexBlob, BytesReference.bytes(repositoryData.snapshotsToXContent(XContentFactory.contentBuilder(XContentType.JSON))), true);
```
Comment (tlrx) [inline server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java]: Nit: you can use XContentFactory.jsonBuilder() and have everything on the same line ;)

**Review by tlrx — APPROVED** [review]
LGTM! I left two minor comments

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/bwc

**Pull request merged**